source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,378 @@
+                                // If the current refinement level is smaller or equal to the refinement minimum level, any coarsening flags should be cleared.
+                                if (cell->level() <= isotherms_i[isotherm_line][0])
+                                  {
+                                    clear_coarsen = true;
+                                  }
+                                // If the current refinement level is smaller then the minimum level, a refinment flag should be placed, and we don't have to look any further.
+                                if (cell->level() <  isotherms_i[isotherm_line][0])
+                                  {
+                                    refine = true;
+                                    break;
+                                  }
+
+                                // If the current refinement level is larger or equal to the maximum refinement level, any refinement flag should be cleared.
+                                if (cell->level() >= isotherms_i[isotherm_line][1])
+                                  {
+                                    clear_refine = true;
+                                  }
+                                // If the current refinement level is larger then the maximum level, a coarsening flag should be placed, and we don't have to look any further.
+                                if (cell->level() >  isotherms_i[isotherm_line][1])
+                                  {
+                                    coarsen = true;
+                                    break;
+                                  }
+
+                              }
+                          }
+                      }
+                  }
+
+                /// Perform the actual placement of the coarsening and refinement flags
+                if (clear_refine)
+                  cell->clear_refine_flag ();
+                if (coarsen)
+                  cell->set_coarsen_flag ();
+                if (clear_coarsen)
+                  cell->clear_coarsen_flag ();
+                if (refine)
+                  cell->set_refine_flag ();
+              }
+        }
+    }
+
+    template <int dim>
+    void
+    Isotherms<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+
+        prm.enter_subsection("Isotherms");
+        {
+
+          prm.declare_entry ("Exclude composition", "-1",
+                             Patterns::Double (),
+                             "Number of mininum and maximum areas defined by isotherms are going to be defined."
+                            );
+          prm.declare_entry ("List of isotherms", "",
+                             Patterns::List(Patterns::List(Patterns::Anything(),0,100000000,","),0,100000000,";"),
+                             "Number of mininum and maximum areas defined by isotherms are going to be defined."
```

Comment:
Both of these parameters could use a more extensive description. I don't think I understand what they are supposed to indicate :-)